src/js/containers/aboutTheData/AgenciesContainer.jsx

```diff
@@ -1,93 +1,276 @@
 import CellWithModal from 'components/aboutTheData/CellWithModal';
-import { agenciesTableColumns, mockAPI } from './AgencyTableMapping';
-
-const dateRows = mockAPI.dates.results
-    .map(({
-        name,
-        abbreviation,
-        code,
-        current_total_budget_authority_amount: total,
-        periods
-    }) => ([
-        (<DrilldownCell data={`${name} (${abbreviation})`} id={code} />),
-        (<div className="generic-cell-content">{total}</div>),
-        ...periods.map(({ date }) => (<div className="generic-cell-content">{date}</div>))
-    ]));
+import { setTableData, setTableSort, setTotals } from 'redux/actions/aboutTheData';
+import { getTotals, getDetails, getDates, usePagination, isPeriodSelectable } from 'helpers/aboutTheDataHelper';
+import DetailsRow from 'models/v2/aboutTheData/BaseAgencyRow';
+import DatesRow from 'models/v2/aboutTheData/DatesRow';
+
+import { agenciesTableColumns } from './AgencyTableMapping';
 
 const propTypes = {
     openModal: PropTypes.func.isRequired,
-    activeTab: PropTypes.oneOf(['dates', 'details'])
+    activeTab: PropTypes.oneOf(['dates', 'details']).isRequired,
+    selectedFy: PropTypes.string,
+    selectedPeriod: PropTypes.string
 };
 
-const AgenciesContainer = ({ activeTab, openModal }) => {
-    const [sortStatus, updateSort] = useState({ field: '', direction: 'asc' });
-    const [{ vertical: isVertialSticky, horizontal: isHorizontalSticky }, setIsSticky] = useState({ vertical: false, horizontal: false });
-    const [loading] = useState(false);
-    const [error] = useState(false);
+const parsePeriods = (periods) => periods
+    .map(({ publicationDate, showNotCertified, isQuarterly }) => (
+        <div className="generic-cell-content">
+            {(!isQuarterly && publicationDate) && publicationDate}
+            {isQuarterly && "Submitted quarterly"}
+            {!publicationDate && "Not submitted"}
+            {showNotCertified && publicationDate && <span className="not-certified">NOT CERTIFIED</span>}
+        </div>
+    ));
+
+const AgenciesContainer = ({
+    activeTab,
+    openModal,
+    selectedFy,
+    selectedPeriod
+}) => {
+    const {
+        details, dates, datesSort, detailsSort, totals, submissionPeriods
+    } = useSelector((state) => ({ ...state.aboutTheData, submissionPeriods: state.account.submissionPeriods }));
+    const dispatch = useDispatch();
+    const datesReq = useRef(null);
+    const detailsReq = useRef(null);
+    const totalsReq = useRef(null);
+    const [
+        [{ page: detailsPage, limit: detailsLimit }, updateDetailsPagination],
+        [{ page: datesPage, limit: datesLimit }, updateDatesPagination]
+    ] = [usePagination(), usePagination()];
+    const [{ vertical: isVerticalSticky, horizontal: isHorizontalSticky }, setIsSticky] = useState({ vertical: false, horizontal: false });
+    const [[, areDetailsLoading, areDatesLoading], setLoading] = useState([true, true, true]);
```

Comment:
Do we want the first item of this array to be empty?

Reply:
the first item, yes